### PR TITLE
Generic name to object

### PR DESCRIPTION
**Pull request by snipercup**
#### Summary
```SUMMARY: none```

#### Purpose of change
Contributes to #36110 and #36611

#### Describe the solution
Change name to object

#### Describe alternatives you've considered
Not changing

#### Testing
image
image
image

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -2647,7 +2647,7 @@
     "category": "spare_parts",
     "symbol": "P",
     "color": "dark_gray",
-    "name": "small lock and key",
+    "name": { "str": "small lock and key" },
```
Comment (snipercup) [inline data/json/items/generic.json]:
```suggestion
    "name": { "str": "small lock and key", "str_pl": "small locks and keys" },
```

File: data/json/items/vehicle/plating.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   {
     "type": "GENERIC",
     "id": "sheet_metal",
-    "name": "sheet metal",
+    "name": { "str": "sheet metal" },
```
Comment (snipercup) [inline data/json/items/vehicle/plating.json]:
```suggestion
    "name": { "str": "metal sheet" },
```

**Comment by snipercup** [comment]
I checked that the name "metal sheet" was not already taken by another item and then changed "sheet metal" to "metal sheet". Please let me know if there is a better way to solve the awkward plural "sheet metals" (as in the screenshot)

**Comment by snipercup** [comment]
Travis complains:
```
/bin/sh: 2: Syntax error: Unterminated quoted string
Makefile:1103: recipe for target 'json_formatter' failed
make: *** [json_formatter] Error 2
```

I'm not sure why it fails, the json is valid and I think the travis error is not caused by my pr.

**Comment by ZhilkinSerg** [comment]
You can just go with `{ "str": "sheet metal", "str_pl": "sheet metal" }`

File: data/json/items/vehicle/plating.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   {
     "type": "GENERIC",
     "id": "sheet_metal",
-    "name": "sheet metal",
+    "name": { "str": "metal sheet" },
```
Comment (snipercup) [inline data/json/items/vehicle/plating.json]:
```suggestion
    "name": { "str": "sheet metal", "str_pl": "sheet metal" },
```

**Comment by snipercup** [comment]
Thanks I fixed the sheet metal

**Pull request merged**